packages/snaps-controllers/src/snaps/SnapController.test.tsx

```diff
@@ -1653,30 +1656,31 @@ describe('SnapController', () => {
       },
     });
 
-    const [snapController, service] = getSnapControllerWithEES(options);
-    const snap = snapController.getExpect(MOCK_SNAP_ID);
-
-    jest
-      // Cast because we are mocking a private property
-      .spyOn(service, 'setupSnapProvider' as any)
-      .mockImplementation((_snapId, rpcStream) => {
-        const mux = setupMultiplex(rpcStream as Duplex, 'foo');
-        const stream = mux.createStream('metamask-provider');
-        const engine = new JsonRpcEngine();
-        const middleware = createAsyncMiddleware(async (req, res, _next) => {
-          if (req.method === 'eth_blockNumber') {
-            await sleep(100);
-            res.result = MOCK_BLOCK_NUMBER;
-          }
-        });
-        engine.push(middleware);
-        const providerStream = createEngineStream({ engine });
-        pipeline(stream, providerStream, stream, (error) => {
-          if (error) {
-            logError(`Provider stream failure.`, error);
-          }
-        });
+    const setupSnapProvider = (_snapId: string, rpcStream: Duplex) => {
+      const mux = setupMultiplex(rpcStream, 'foo');
+      const stream = mux.createStream('metamask-provider');
+      const engine = new JsonRpcEngine();
+      const middleware = createAsyncMiddleware(async (req, res, _next) => {
+        if (req.method === 'eth_blockNumber') {
+          await sleep(100);
```

Comment:
I see we were doing this before, but what's the reason for it?

Reply:
The test is testing the behavior that we don't time out requests while the Snap is waiting for MetaMask. So the request timeout is super low and we emulate MetaMask taking a "long" time to respond to check that the Snap isn't terminated for waiting on it